front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -97,34 +98,70 @@ const DEFAULT_PAYLOAD: Pick<
   start_time: '2024-07-15T08:00:00+02:00',
 };
 
-const createPathItemFromNode = (node: Node, index: number) => {
+const createPathItemFromNode = async (
+  node: Node,
+  index: number,
+  infraId: number,
+  dispatch: AppDispatch
+) => {
   const [trigram, secondaryCode] = node.betriebspunktName.split('/');
-  return {
+  let finalSecondaryCode: string | undefined;
+  if (!secondaryCode) {
+    const searchPayload = {
+      object: 'operationalpoint',
+      query: ['and', ['=', ['infra_id'], infraId], ['=', ['trigram'], trigram]],
+    };
+    let searchResults = (await dispatch(
+      osrdEditoastApi.endpoints.postSearch.initiate({
+        searchPayload,
+      })
+    ).unwrap()) as SearchResultItemOperationalPoint[];
+
+    let opFound = searchResults.find((op) => op.ch === 'BV' || op.ch === '00');
+    if (!opFound) opFound = searchResults.find((op) => op.ch === '');
+    finalSecondaryCode = opFound?.ch;
```

Comment:
Is it necessary to split the `find` operation into two parts? We could simplify it by merging them like this:

```suggestion
   finalSecondaryCode = searchResults.find((op) => op.ch === 'BV' || op.ch === '00' || op.ch === '')?.ch;
```

Reply:
Yes, I think it's necessary. We want to be able to prioritize the secondary_code, and writing it this way allows us to ensure we have a BV or 00 before the empty string.